vm/src/dictdatatype.rs

```diff
@@ -189,6 +207,7 @@ impl<T> DictInner<T> {
             entries_size: self.entries.len(),
             used: self.used,
             filled: self.filled,
+            nentires: self.nentries,
```

Comment:
this field was added to make popitem fast. if there is no nentries it costs O(N^2) for popping all elements of the dict with n entries. [..., None, None, None] kind of entries cause inefficient work without nentries.

if entry is like [..,valid entry, None, None, None] , then nentries points to the left None, popitem would pick the valid entry at once. also, if i insert one entry, this would make the entires to [..,valid entry, valid entry, None, None] which make popitem remains fast

Reply:
I think you are talking about `DictInner::nentries`. I agree that's a clever way to do that. But this one is `DictSize::nentires`.

Reply:
ah, that's not useful now. but i thought that size function should give the size information about the dict and nentries is about the valid entries length, so i included on the size function. if it is better to remove the unused field, then i'll remove it

Reply:
Let's go without it for for now. I think we can add it when we actually can use it later. It will be straightforward because it doesn't have any trick but just copying the field.